sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpProxySelector.java

```diff
@@ -34,7 +34,7 @@ public OkHttpProxySelector(Proxy.Type proxyType, SocketAddress proxyAddress, Str
          * If the host of the URI matches the nonProxyHostsPattern return no options for proxying, otherwise return the
          * proxy.
          */
-        return (nonProxyHostsPattern == null || !nonProxyHostsPattern.matcher(uri.getHost()).matches())
+        return (nonProxyHostsPattern == null || !nonProxyHostsPattern.matcher(uri.getHost()).find())
```

Comment:
Why did we change to partial match here instead of whole string match?

Reply:
This should be reverted back to `find` as the pattern should match the entire regex

Reply:
Did you mean reverted back to `matches()`?

Reply:
Yes 😃